evap/staff/importers.py

```diff
@@ -350,15 +357,16 @@ def check_data_type_correctness(self):
         """
         Checks that all cells after the skipped rows contain string values (not floats or integers).
         """
-        for sheet in self.book.sheets():
-            for row in range(self.skip_first_n_rows, sheet.nrows):
-                if not all(isinstance(cell, str) for cell in sheet.row_values(row)):
-                    self.errors[ImporterError.SCHEMA].append(
-                        _(
-                            "Wrong data type in sheet '{}' in row {}."
-                            " Please make sure all cells are string types, not numerical."
-                        ).format(sheet.name, row + 1)
-                    )
+        for sheet in self.book:
+            for row_idx, row in enumerate(sheet.iter_rows(values_only=True)):
+                for cell in row:
+                    if cell and not isinstance(cell, str):
```

Comment:
the `all(...)` approach before had the advantage that we wouldn't get multiple errors per row, so I'd try to keep `any()` or `all()` here.